File: 프로그래머스/lv1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool chk[32] = { false, };

int solution(int n, vector<int> lost, vector<int> reserve) {
    int ret = n;

    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());

    for(int i=0; i<lost.size(); i++)
        for (int j = 0; j < reserve.size(); j++) {
            if (lost[i] == reserve[j]) {
                reserve.erase(reserve.begin() + j);
                lost.erase(lost.begin() + i);
                i--;
                break;
            }
        }

    for (int i = 0; i < reserve.size(); i++)
        chk[reserve[i]] = true;

    for (int i = 0; i < lost.size(); i++) {
        int cur = lost[i];
        
        if (chk[cur - 1])
            chk[cur] = false;
        else if (chk[cur + 1])
            chk[cur + 1] = false;
        else
            ret--;
    }
    return ret;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    check = [1 for _ in range(n + 2)]
    check[n + 1] = 0
    check[0] = 0

    for x in reserve:
        check[x] = 2
    
    for x in lost:
        check[x] = check[x] - 1

    for i in range(1, n + 1):
        if(check[i] == 0):
            if check[i - 1] > 1:
                check[i - 1] -= 1
                check[i] = 1
            elif check[i + 1] > 1:
                check[i + 1] -= 1
                check[i] = 1

    for i in range(1, n + 1):
        answer += check[i] > 0
    
    return answer